js/form.js

```diff
@@ -0,0 +1,134 @@
+console.clear();
+
+const form = document.querySelector('[data-js="question-form"]');
+const mainContainer = document.querySelector("main");
+const inputQuestion = document.querySelector('[data-js="question-form-input"]');
+const characterCounterQuestion = document.querySelector(
+  '[data-js="character-counter-question"]'
+);
+const inputAnswer = document.querySelector('[data-js="answer-form-input"]');
+const characterCounterAnswer = document.querySelector(
+  '[data-js="character-counter-answer"]'
+);
+
+inputQuestion.focus();
+
+// This is the function for the character counters
+
+function textfieldInput(input, characterCount) {
+  characterCount.textContent =
+    input.getAttribute("maxlength") - input.value.length;
+}
+
+inputQuestion.addEventListener("input", () => {
+  textfieldInput(inputQuestion, characterCounterQuestion);
+});
+
+inputAnswer.addEventListener("input", () => {
+  textfieldInput(inputAnswer, characterCounterAnswer);
+});
+
+// Function to add '#'
+
+function hashtag(data) {
+  const input = data.split(" ").join("");
+  if (input.startsWith("#")) {
+    return input;
+  } else {
+    return "#" + input;
+  }
+}
+
+// Here starts the function for form submission and card creation/
+
+form.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const formData = new FormData(event.target);
+  const data = Object.fromEntries(formData);
+
+  // Declaring input variables
+
+  const questionInput = data.question;
```

Comment:
Ah, hier hab ich direkt die Antwort. Also Du benutzt es. Aber ich frag mich, ob das sinnvoll ist. Du hast ja oben die Variablen definiert (zumindest inputQuestion und inputAnswer). Dann müsste es doch einfacher gehen, einfach zu schreiben
```suggestion
  const questionInput = inputQuestion.value;
```
oder? Und dann brauchst Du das ganze Objekt nicht?  Vielleicht übersehe ich etwas, aber das sieht für mich einfacher aus. (Die Benennung mit inputQuestion und questionInput finde ich übrigens auch etwas verwirrend, aber das ist was anderes. :grin: